cardano-testnet/src/Test/Assert.hs

```diff
@@ -0,0 +1,104 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+{- HLINT ignore "Redundant return" -}
+
+module Test.Assert
+  ( readJsonLines
+  , assertChainExtended
+  , getRelevantLeaderSlots
+  ) where
+
+import           Control.Applicative ((<*>))
+import           Control.Monad (Monad (..))
+import           Control.Monad.IO.Class (MonadIO)
+import           Control.Monad.Trans.Reader (ReaderT)
+import           Control.Monad.Trans.Resource (ResourceT)
+import           Data.Aeson (FromJSON (..), Value, (.:))
+import           Data.Bool (Bool (..))
+import           Data.Eq (Eq (..))
+import           Data.Function (($), (.))
+import           Data.Functor ((<$>))
+import           Data.Int (Int)
+import           Data.Maybe (Maybe (..), mapMaybe)
+import           Data.Ord (Ord (..))
+import           Data.Text (Text)
+import           GHC.Stack (HasCallStack)
+import           Hedgehog (MonadTest)
+import           Hedgehog.Extras.Internal.Test.Integration (IntegrationState)
+import           System.FilePath (FilePath)
+import           System.IO (IO)
+import           Test.Runtime (NodeLoggingFormat (..))
+import           Text.Show (Show (..))
+
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Types as Aeson
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.List as L
+import qualified Data.Maybe as Maybe
+import qualified Data.Time.Clock as DTC
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Stock.IO.File as IO
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Test.Process as H
+
+readJsonLines :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m [Value]
+readJsonLines fp = mapMaybe (Aeson.decode @Value) . LBS.split 10 <$> H.evalIO (LBS.readFile fp)
+
+fileJsonGrep :: FilePath -> (Value -> Bool) -> IO Bool
+fileJsonGrep fp f = do
+  lines <- LBS.split 10 <$> LBS.readFile fp
+  let jsons = mapMaybe (Aeson.decode @Value) lines
+  return $ L.any f jsons
+
+assertChainExtended :: (H.MonadTest m, MonadIO m)
+  => DTC.UTCTime
+  -> NodeLoggingFormat
+  -> FilePath
+  -> m ()
+assertChainExtended deadline nodeLoggingFormat nodeStdoutFile =
+  H.assertByDeadlineIOCustom "Chain not extended" deadline $ do
+    case nodeLoggingFormat of
+      NodeLoggingFormatAsText -> IO.fileContains "Chain extended, new tip" nodeStdoutFile
```

Comment:
Out of curiosity, do we want to maintain the parsing of plain text?

Reply:
Not necessarily, but I'm keeping it so we don't need to do a big bang migration.  Once we're satisfied we don't use it anywhere we can remove it.